plugins/Referrers/VisitorDetails.php

```diff
@@ -38,6 +38,19 @@ public function renderVisitorDetails($visitorDetails)
         return [[ 10, $view->render() ]];
     }
 
+    public function renderActionTooltip($action, $visitInfo)
+    {
+        if ($action['type'] !== 'goal' && $action['type'] !== 'ecommerceOrder') {
+            return [];
+        }
+
+        // Attribution information for goals
+        $view         = new View('@Referrers/_actionTooltip');
```

Comment:
Should this information be completely hidden if no referrerType was detected?

One of my local test goal conversions was created by the VisitorGenerator (`--generate-visits-db`) and has `referrer_type == NULL` in the database. So no real data, but most of the `VisitorDetails::getReferrerType()` implementations can technically return an empty string.

Reply:
Guess it won't hurt to check that. Even though it shouldn't occur.